core/sodasql/scan/warehouse.py

```diff
@@ -14,16 +14,18 @@
 from sodasql.scan.db import sql_fetchone, sql_fetchall, sql_fetchone_description, sql_fetchall_description
 from sodasql.scan.dialect import Dialect
 from sodasql.scan.warehouse_yml import WarehouseYml
+from sodasql.telemetry.soda_telemetry import soda_telemetry
 
 logger = logging.getLogger(__name__)
 
-
 class Warehouse:
 
     def __init__(self, warehouse_yml: WarehouseYml):
         self.name = warehouse_yml.name
         self.dialect: Dialect = warehouse_yml.dialect
         self.connection = self.dialect.create_connection()
+        soda_telemetry.set_attribute('datasource_type', self.dialect.type)
+        soda_telemetry.set_attribute('datasource_hash', soda_telemetry.obtain_datasource_hash(self.dialect))
```

Comment:
In #543 (haha it's almost the default postgres port 😂 ) there is no attribute named `datasource_hash`. Did you mean `datasource_id`? If so, then I think we're sending the wrong thing to it.

Sorry for the stupid question. What is the difference between `dialect.type` and `dialect`?

Reply:
Ok thanks for the clarification. I was 99% sure of that after seeing more of the code.

So I think for me, it's a case of making sure we don't hash the entire dialect object into the `datasource_hash` that will become the `datasource_id` and hopefully my comments on each dialect about what should be hashed form the dialect object into `datasource_id` should get us in a really good shape.